File: 2020-week9/cpp/2020-week9_2.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{

    int n;
    scanf("%d", &n);

    int count = 0;
    int num = 666;

    while (count < n)
    {
        string str = to_string(num);

        for (int i = 0; i <= str.length() - 1; i++)
        {
            bool isPass = true;

            for (int k = 0; k < 3; k++)
            {
                if (str[i + k] != '6')
                {
                    isPass = false;
                    break;
                }
            }

            if (isPass)
            {
                count++;
                break;
            }
        }
        num++;
    }
    printf("%d\n", num - 1);

    return 0;
}

File: 2020-week9/java-jh/2020-week9-2.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		int num = 666;
		int count = 0;
		while (count < n) {
			String numStr = Integer.toString(num);
			if (numStr.contains("666"))
				count++;
			num++;
		}
		System.out.print(num - 1);
	}
}